source/plugins/ruby/in_kube_nodes.rb

```diff
@@ -496,4 +501,98 @@ def getNodeTelemetryProps(item)
       return properties
     end
   end # Kube_Node_Input
+
+
+  class NodeStatsCache
+    # inner class for caching implementation (CPU and memory caching is handled the exact same way, so logic to do so is moved to a private inner class)
+    # (to reduce code duplication)
+    class NodeCache
+
+      @@RECORD_TIME_TO_LIVE = 60*20  # units are seconds, so clear the cache every 20 minutes.
+  
+      def initialize
+        @cacheHash = {}  # 
+        @timeAdded = {}  # records when an entry was last added
+        @lock = Mutex.new
+        @lastCacheClearTime = 0
+  
+        @cacheHash.default = 0.0
+        @lastCacheClearTime = DateTime.now.to_time.to_i
+      end
+  
+      def get_capacity(node_name)
+        begin
+          @lock.lock
+          retval = @cacheHash[node_name]
+          return retval
+        ensure
+          # this is always run even though there was a return earlier
+          @lock.unlock
+        end
+      end
+  
+      def set_capacity_all(capacity_json)
+        # clean the cache here so that calling code doesn't have to remember to clean it. The disadvantage to this 
+        # approach is that the cache will never be cleaned if new nodes are never added, (but then the cache also 
+        # can't grow and our main concern is unbounded cache growth)
+        if DateTime.now.to_time.to_i - @lastCacheClearTime > @@RECORD_TIME_TO_LIVE
```

Comment:
You are invoking set_capacity_all for every minute with list of nodes available atm,. in that case, just taking that snapshot override the existing cache with new list isnt that good enough? why we need to maintain cache for non-existent nodes?